src/providers/Elsa.DistributedLocking.AzureBlob/AzureBlobLockProvider.cs

```diff
@@ -0,0 +1,199 @@
+using Elsa.DistributedLock;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+using Microsoft.Azure.Storage.RetryPolicies;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Elsa.DistributedLocking.AzureBlob
+{
+    public class AzureBlobLockProvider : IDistributedLockProvider
+    {
+        private const string Prefix = "elsa";
+        private const string ContainerName = "elsa-lock-container";
+        private readonly AutoResetEvent _leaseSemaphore = new AutoResetEvent(true);
+        private static readonly object syncRoot = new Object();
+        private readonly List<LockedBlob> _lockedBlobs = new List<LockedBlob>();
+        private readonly string _connectionString;
+        private TimeSpan _leaseTime;
+        private readonly ILogger<AzureBlobLockProvider> _logger;
+        private CloudBlobContainer _cloudBlobContainer;
+        private readonly TimeSpan _renewInterval;
+        private Timer _renewTimer;
+
+        public AzureBlobLockProvider(string connectionString, TimeSpan leaseTime, TimeSpan renewInterval,
+                                        ILogger<AzureBlobLockProvider> logger)
+        {
+            _logger = logger;
+            _connectionString = connectionString;
+            if (leaseTime > TimeSpan.FromSeconds(60) || leaseTime < TimeSpan.FromSeconds(15))
```

Comment:
As per this [documentation](https://docs.microsoft.com/en-us/rest/api/storageservices/lease-blob), the lock duration can be 15 to 60 seconds, or can be infinite and hence the Guard. 
If it is taking more than 60 seconds to process the next block it will automatically try to re-acquire a lock.

Reply:
I see. I didn't know. Would you mind adding a brief comment referencing this documentation? And also move the literal values into a constant field? Thanks!

Reply:
As suggested I have added the reference link and moved literal values to constant field, Now this PR should be ready to merge! @sfmskywalker